server/controllers/chartsController.js

```diff
@@ -145,14 +145,15 @@ export const graphDataController = async (dataDb, userAccess, chart_id) => {
     chart.fieldLabelValueMap.forEach((fieldLabelValueMap) => {
       const { color, label, value, targetValues } = fieldLabelValueMap
       const targetValue = targetValues[study]
-      const hasValue = subjectDayData.some(
-        (dayData) => dayData[chart.variable] == value
-      )
+      const isVariableValueEmpty = value === ''
+      const canSubjectDataBeCounted = !isVariableValueEmpty
+        ? subjectDayData.some((dayData) => dayData[chart.variable] == value)
+        : subjectDayData.every((dayData) => dayData[chart.variable] == value)
```

Comment:
I think we should flip this to avoid the double negative.

```suggestion
      const canSubjectDataBeCounted = isVariableValueEmpty
        ? subjectDayData.every((dayData) => dayData[chart.variable] == value)
        : subjectDayData.some((dayData) => dayData[chart.variable] == value)
```

Reply:
do we also want to count the subject if the `subjectDayData` is empty?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/199/commits/9ead0ec842599f647148d5e5f9e8ac0cc04912cf